File: trabalho-02/client/services/chats_service.py
import requests

class ChatsService():

    def __init__(self, url):
        self.url = url

    
    def load_chats_info(self, token):
        try:
            response = requests.get(
                f'{self.url}/chats',
                headers={'Authorization': f'Bearer {token}'}
            )
            return response.json()
        except Exception as e:
            print(e)
            return {'error': str(e)}
    
    def load_chat_details(self, token, chat_id):
        try:
            response = requests.get(
                f'{self.url}/chats/{chat_id}',
                headers={'Authorization': f'Bearer {token}'}
            )
            return response.json()
        except Exception as e:
            print(e)
            return {'error': str(e)}
    
    def create_chat(self, token, name, chat_type):
        try:
            response = requests.post(
                f'{self.url}/chats',
                headers={'Authorization': f'Bearer {token}'},
                json={'name': name, 'chat_type': chat_type}
            )
            return response.json()
        except Exception as e:
            print(e)
            return {'error': str(e)}

File: trabalho-02/clients/react_client/chatuba_web_client/src/services/ChatsService.js
import axios from 'axios';

export async function LoadChatsInfo(token) {
    const { data } = await axios.get('http://localhost:8080/api/chats', {
        headers: {
            'Content-Type': 'application/json',
            'Authorization': `Bearer ${token}`
        }
    })

    return data
};

export async function LoadChatInfo(token, chatId) {
    const { data } = await axios.get(`http://localhost:8080/api/chats/${chatId}`, {
        headers: {
            'Content-Type': 'application/json',
            'Authorization': `Bearer ${token}`
        }
    })

    return data;
}